tests/NCronJob.Tests/NCronJobIntegrationTests.cs

```diff
@@ -11,40 +11,49 @@ public sealed class NCronJobIntegrationTests : JobIntegrationBase
     [Fact]
     public async Task CronJobThatIsScheduledEveryMinuteShouldBeExecuted()
     {
-        ServiceCollection.AddNCronJob(n => n.AddJob<SimpleJob>(p => p.WithCronExpression(Cron.AtEveryMinute)));
-        var provider = CreateServiceProvider();
+        ServiceCollection.AddNCronJob(n => n.AddJob<DummyJob>(p => p.WithCronExpression(Cron.AtEveryMinute)));
 
-        await provider.GetRequiredService<IHostedService>().StartAsync(CancellationToken);
+        (IDisposable subscription, IList<ExecutionProgress> events) = RegisterAnExecutionProgressSubscriber(ServiceProvider);
+
+        await ServiceProvider.GetRequiredService<IHostedService>().StartAsync(CancellationToken);
 
         FakeTimer.Advance(TimeSpan.FromMinutes(1));
-        var jobFinished = await WaitForJobsOrTimeout(1);
-        jobFinished.ShouldBeTrue();
+
+        var orchestrationId = events[0].CorrelationId;
+
+        await WaitForOrchestrationCompletion(events, orchestrationId);
+
+        subscription.Dispose();
+
+        var filteredEvents = events.FilterByOrchestrationId(orchestrationId);
+        filteredEvents.ShouldBeScheduledThenCompleted();
     }
 
     [Fact]
     public async Task AdvancingTheWholeTimeShouldHaveTenEntries()
     {
-        ServiceCollection.AddNCronJob(n => n.AddJob<SimpleJob>(p => p.WithCronExpression(Cron.AtEveryMinute)));
-        var provider = CreateServiceProvider();
+        ServiceCollection.AddNCronJob(n => n.AddJob<DummyJob>(p => p.WithCronExpression(Cron.AtEveryMinute)));
+
+        (IDisposable subscription, IList<ExecutionProgress> events) = RegisterAnExecutionProgressSubscriber(ServiceProvider);
 
-        await provider.GetRequiredService<IHostedService>().StartAsync(CancellationToken);
+        await ServiceProvider.GetRequiredService<IHostedService>().StartAsync(CancellationToken);
 
         void AdvanceTime() => FakeTimer.Advance(TimeSpan.FromMinutes(1));
-        var jobFinished = await WaitForJobsOrTimeout(10, AdvanceTime);
 
-        jobFinished.ShouldBeTrue();
-    }
+        AdvanceTime();
 
-    [Fact]
-    public async Task JobsShouldCancelOnCancellation()
```

Comment:
@linkdotnet Did that (not migrated) test do something that `RetryTests.JobShouldHonorCancellation()` doesn't already cover?

Reply:
I do think JobShouldHonorCancellation is more generic in the sense that it is the outside system triggering the cancellation. Which is better.

If I look at JobsShouldCancelOnCancellation I am not really sure what it does. It tests test-logic but absolutely zero prod code.